Example/src/ScrollToExample.js

```diff
@@ -0,0 +1,153 @@
+import React, { useRef, useEffect } from 'react';
+import { StyleSheet, View, Text, ScrollView } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  useAnimatedGestureHandler,
+  scrollTo,
+  getTag,
+  useDerivedValue,
+} from 'react-native-reanimated';
+import { PanGestureHandler } from 'react-native-gesture-handler';
+
+const digits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+const indices = [0, 1, 2, 3];
+
+const range = [0, 9999];
+
+const dotSize = 40;
+
+export default function ScrollToScreen() {
+  const progress = useSharedValue(0);
+  const number = useDerivedValue(() => {
+    const val = range[0] + Math.round(progress.value * (range[1] - range[0]));
+    return val;
+  });
+
+  return (
+    <SafeAreaView>
+      <View style={{ alignItems: 'center' }}>
+        <NumberDisplay number={number} />
+        <Text>move dot</Text>
+        <View>
+          <ProgressBar progress={progress} />
+        </View>
+      </View>
+    </SafeAreaView>
+  );
+}
+
+function getDigit(number, i) {
+  return useDerivedValue(() => {
+    return Math.floor(number.value / 10 ** i) % 10;
+  });
+}
+
+function NumberDisplay({ number }) {
+  return (
+    <View style={{ height: 400, width: 200 }}>
+      <View
+        style={{
+          flexDirection: 'row-reverse',
+          justifyContent: 'space-between',
+          alignItems: 'center',
+        }}>
+        {indices.map((i) => {
+          return <Digit digit={getDigit(number, i)} key={i} />;
+        })}
+      </View>
+    </View>
+  );
+}
+
+function Digit({ digit }) {
+  const tag = useSharedValue(-1);
```

Comment:
I was thinking of a slightly different API that'd allow us to avoid some of the boilerplate here:

```js
const ref = useAnimatedRef();
```

Then in view we could just do:
```js
<ScrollView ref={ref}>
```

And in the worklet we could use it this way:
```js
scrollTo(ref, 0, difit.value * 200, true);
```

This would be the easiest approach in my opinion and could be implemented in the follwoing way:
`useAnimatedRef` should return a function – refs can be functions that are called when the view ref is set. When that happens the function should update a shared values kept underneath. Methods like `scrollTo` or `measure` should expect `ref` to be an object and extract the reference to the shared value out of it.

Reply:
The only problem with that approach - you may need this ref as a regular ref somewhere else too.

Reply:
Here is the sample solution
https://github.com/software-mansion/react-native-reanimated/pull/1040

Reply:
This works although it might be good to consider whether there would be any more cases of using this, also as @terrysahaidak mentioned the name could be misleading.